Sources/NIOSOCKS/Channel Handlers/SOCKSServerHandshakeHandler.swift

```diff
@@ -102,15 +108,21 @@ public final class SOCKSServerHandshakeHandler: ChannelDuplexHandler, RemovableC
     }
     
     private func handleWriteAuthenticationData(_ data: ByteBuffer, complete: Bool, context: ChannelHandlerContext, promise: EventLoopPromise<Void>?) throws {
-        do {
-            try self.stateMachine.sendData()
-            if complete {
+        
```

Comment:
Something that I have seen just now. A function that throws and takes a promise seems very odd to me from an API view. Yes this is only a private method... but I think some rules even apply for private methods.

Reply:
I've refactored, see what you think :)

Reply:
Looks good!